packages/docusaurus-plugin-sitemap/src/__tests__/options.test.ts

```diff
@@ -49,4 +50,17 @@ describe('validateOptions', () => {
       `"\\"changefreq\\" must be one of [daily, monthly, always, hourly, weekly, yearly, never]"`,
     );
   });
+
+  it('rejects bad ignorePatterns inputs', () => {
+    expect(() =>
+      testValidate({ignorePatterns: /^search/}),
+    ).toThrowErrorMatchingInlineSnapshot(
+      `"\\"ignorePatterns\\" must be an array"`,
+    );
+    expect(() =>
+      testValidate({ignorePatterns: ['search']}),
+    ).toThrowErrorMatchingInlineSnapshot(
+      `"\\"ignorePatterns[0]\\" must be of type object"`,
```

Comment:
I _think_ we can make it nicer if we add `.items(Joi.object().instance(RegExp).messages({ "object.instance": "{#path} must be a regex" ))`. Joi is extremely hard to play with, so if you don't want to bother, I can take a look later.